### PR TITLE
Add phpdoc for Email and EmailHandler

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #358

### CODE REVIEW

**Comment by samdark** [comment]
Arrgh. Wrong branch was used as a base. Review only Email and EmailHandler please.

File: src/Rule/Email.php [hunk 1]
```diff
@@ -30,61 +29,51 @@ final class Email implements RuleWithOptionsInterface, SkipOnErrorInterface, Whe
     use SkipOnErrorTrait;
```
Comment (arogachev) [inline src/Rule/Email.php:29]: Description:

> Validates that the value is a valid email address.

But rule does not perform the actual validation, it's done by handler.

File: src/Rule/Email.php [hunk 2]
```diff
@@ -30,61 +29,51 @@ final class Email implements RuleWithOptionsInterface, SkipOnErrorInterface, Whe
...
+     *
+     * - `{attribute}`: the label of the attribute being validated.
+     * - `{value}`: the value of the attribute being validated.
+     * @param bool|callable|null $skipOnEmpty Whether to skip this rule if the value validated is empty. See {@see SkipOnEmptyInterface}.
+     * @param bool $skipOnError Whether to skip this rule if any of the previous rules gave an error. See {@see SkipOnErrorInterface}.
+     * @param Closure|null $when A callable to define a condition for applying the rule. See {@see WhenInterface}.
+     * @psalm-param WhenType $when
+     */
     public function __construct(
-        #[Language('RegExp')]
```
Comment (arogachev) [inline src/Rule/Email.php:34]: Why language attributes were removed?

File: src/Rule/Email.php [hunk 3]
```diff
@@ -30,61 +29,51 @@ final class Email implements RuleWithOptionsInterface, SkipOnErrorInterface, Whe
...
+     * @param string $fullPattern The regular expression used to validate email addresses with the name part.
+     * This property is used only when {@see $allowName} is `true`.
+     * @param string $idnEmailPattern The regular expression used to validate complex emails when {@see $enableIdn} is `true`.
+     * @param bool $allowName bool Whether to allow a name in the email address (e.g. "John Smith <john.smith@example.com>").
+     * Defaults to `false`. See {@see $fullPattern}.
+     * @param bool $checkDns bool Whether to check email's domain exists and has either an A or MX record.
+     * Be aware that this check can fail due to temporary DNS problems even if the email address is
+     * valid and an email would be deliverable. Defaults to `false`.
+     * @param bool $enableIdn Whether validation process should take IDN (internationalized domain names) into account.
+     * Defaults to false meaning that validation of emails containing IDN will always fail.
```
Comment (arogachev) [inline src/Rule/Email.php:43]:
```suggestion
     * Defaults to `false` meaning that validation of emails containing IDN will always fail.
```

File: src/Rule/Email.php [hunk 4]
```diff
@@ -30,61 +29,51 @@ final class Email implements RuleWithOptionsInterface, SkipOnErrorInterface, Whe
...
+     * Note that in order to use IDN validation you have to install and enable `intl` PHP extension,
+     * otherwise an exception would be thrown.
```
Comment (arogachev) [inline src/Rule/Email.php:45]:
```suggestion
     * otherwise an exception will be thrown.
```

File: src/Rule/Email.php [hunk 5]
```diff
@@ -30,61 +29,51 @@ final class Email implements RuleWithOptionsInterface, SkipOnErrorInterface, Whe
...
-        private bool $enableIDN = false,
+        private bool $checkDns = false,
+        private bool $enableIdn = false,
         private string $incorrectInputMessage = 'The value must have a string type.',
         private string $message = 'This value is not a valid email address.',
-
-        /**
-         * @var bool|callable|null
-         */
         private $skipOnEmpty = null,
```
Comment (arogachev) [inline src/Rule/Email.php:72]:
```suggestion
        private mixed $skipOnEmpty = null,
```

**Review by arogachev — CHANGES_REQUESTED** [review]
See suggestions

File: src/Rule/Email.php (hunk 2 above)
Comment (vjik) [reply src/Rule/Email.php:34]: it's cool feature of PhpStorm

File: src/Rule/Email.php [hunk 6]
```diff
@@ -30,61 +29,51 @@ final class Email implements RuleWithOptionsInterface, SkipOnErrorInterface, Whe
...
-        /**
-         * @var WhenType
-         */
         private Closure|null $when = null,
     ) {
-        if ($enableIDN && !function_exists('idn_to_ascii')) {
+        if ($enableIdn && !function_exists('idn_to_ascii')) {
             // Tested via separate CI configuration (see ".github/workflows/build.yml").
             // @codeCoverageIgnoreStart
             throw new RuntimeException('In order to use IDN validation intl extension must be installed and enabled.');
```
Comment (arogachev) [inline src/Rule/Email.php:79]: Add "throws" tag?

File: src/Rule/Email.php (hunk 2 above)
Comment (samdark) [reply src/Rule/Email.php:34]: It doesn't really work here.

**Pull request merged**